### PR TITLE
Implementation of data seasons

### PR DESCRIPTION
**Pull request by specktakel**
Added some classes to handle analysis/simulation/detector spanning multiple seasons. Class `Uptime` can calculate uptimes during periods, convert start and end times to live time.

### CODE REVIEW

**Comment by minevertokenhire** [comment]
looks good!

File: icecube_tools/point_source_likelihood/point_source_likelihood.py [hunk 1]
```diff
@@ -618,6 +720,165 @@ def get_test_statistic(self):
...
+
+    def _minimize(self):
+        """
+        Minimize -log(likelihood_ratio) for the source hypothesis,
+        returning the best fit ns and index.
+
+        Uses the iMinuint wrapper.
+        """
+
+        init_index = 2.19  # self._energy_likelihood._min_index + (self._max_index - self._energy_likelihood._min_index)/2
```
Comment (cescalara) [inline icecube_tools/point_source_likelihood/point_source_likelihood.py:823]: Do we want this to be hard-coded as such?

File: icecube_tools/point_source_likelihood/point_source_likelihood.py [hunk 2]
```diff
@@ -618,6 +720,165 @@ def get_test_statistic(self):
...
+        m.migrad()
+        if self.which != 'spatial':
+            if not m.fmin.is_valid or not m.fmin.has_covariance:
+
+                # Fix the index as can be uninformative
+                m.fixed["index"] = True
+                m.migrad()
+        else:
+            pass
+            #what else to do?
```
Comment (cescalara) [inline icecube_tools/point_source_likelihood/point_source_likelihood.py:873]: Maybe raise an error or warning here?

**Review by cescalara — COMMENTED** [review]
Could the files in docs/data be merged into a single file?

Thanks for this! I like the new interface and the new docs/tests too.

File: icecube_tools/point_source_likelihood/point_source_likelihood.py (hunk 1 above)
Comment (specktakel) [reply icecube_tools/point_source_likelihood/point_source_likelihood.py:823]: Wasn't the initial value hardcoded before, too?

File: icecube_tools/point_source_likelihood/point_source_likelihood.py (hunk 2 above)
Comment (specktakel) [reply icecube_tools/point_source_likelihood/point_source_likelihood.py:873]: good idea, will do so

File: icecube_tools/point_source_likelihood/point_source_likelihood.py (hunk 1 above)
Comment (cescalara) [reply icecube_tools/point_source_likelihood/point_source_likelihood.py:823]: Well, yes and no. One could argue that 2.19 could be outside the range of _min_index and _max_index

**Comment by specktakel** [comment]
Re: too many files.

I could include that in the event classes in the other branch. Then create only one file with sub-directories (or whatever these are called in h5py) for each index, and when hand each directory's data to the appropriate instance requiring the data.

**Comment by cescalara** [comment]
> I could include that in the event classes in the other branch. Then create only one file with sub-directories (or whatever these are called in h5py) for each index, and when hand each directory's data to the appropriate instance requiring the data.

Sounds good

**Review by cescalara — APPROVED** [review]

**Pull request merged**